src/boost_histogram/_internal/hist.py

```diff
@@ -779,10 +819,123 @@ def __setitem__(self, index, value):
+        }:
+            return Kind.MEAN
+        else:
+            return Kind.COUNT
+
+    def values(self, flow=False):
+        # type: (bool) -> ArrayLike
+        """
+        Returns the accumulated values. The counts for simple histograms, the
+        sum of weights for weighted histograms, the mean for profiles, etc.
+
+        If counts is equal to 0, the value in that cell is undefined if
+        kind == "MEAN".
+
+        :param flow: Enable flow bins. Not part of PlottableHistogram, but
+        included for consistency with other methods and flexibility.
+
+        :return: np.ndarray[np.float64]
+        """
+
+        view = self.view(flow)
+        if len(view.dtype) == 0:
+            return view
+        else:
+            return view.value
+
+    def variances(self, flow=False):
+        # type: (bool) -> Optional[ArrayLike]
+        """
+        Returns the estimated variance of the accumulated values. The sum of squared
+        weights for weighted histograms, the variance of samples for profiles, etc.
+        For an unweighed histogram where kind == "COUNT", this should return the same
+        as values if the histogram was not filled with weights, and None otherwise.
+        If counts is equal to 1 or less, the variance in that cell is undefined if
+        kind == "MEAN".
+
+        If kind == "MEAN", the counts can be used to compute the error on the mean
+        as sqrt(variances / counts), this works whether or not the entries are
+        weighted if the weight variance was tracked by the implementation.
+
+        Currently, this always returns - but in the future, it will return None
+        if a weighted fill is made on a unweighed storage.
+
+        :param flow: Enable flow bins. Not part of PlottableHistogram, but
+        included for consistency with other methods and flexibility.
+
+        :return: np.ndarray[np.float64]
+        """
+
+        view = self.view(flow)
+        if len(view.dtype) == 0:
+            if self._variance_known:
+                return view
+            else:
+                return None
+        else:
+            return view.variance
+
+    def counts(self, flow=False):
+        # type: (bool) -> Optional[ArrayLike]
+        """
+        Returns the number of entries in each bin for an unweighted
+        histogram or profile and an effective number of entries (defined below)
+        for a weighted histogram or profile. An exotic generalized histogram could
+        have no sensible .counts, so this is Optional and should be checked by
+        Consumers.
+
+        For a weighted histogram, counts is defined as sum_of_weights ** 2 /
+        sum_of_weights_squared. It is equal or less than the number of times
+        the bin was filled, the equality holds when all filled weights are equal.
+        The larger the spread in weights, the smaller it is, but it is always 0
+        if filled 0 times, and 1 if filled once, and more than 1 otherwise.
```

Comment:
Add "It can be used to compute the error on the mean like the ordinary counts, if kind == MEAN".